### PR TITLE
feat: Implement new Final Registration (FR) Page

### PR DESCRIPTION
**Pull request by ferran98campos**
## Describe your changes

The user registration page 'https://register.armada.nu/register/user' style has been changed in order to make it more accessible since people noticed the previous version confused some clients.

- The form is now a stepped form that allows users to keep track of the different steps the form has, and in which one they are currently in.
- For them to select a company, a company browser has been developed to replace the old dropdown menu which contained 1000+ companies. Now, some companies will show up (10 companies MAX) that match what they wrote in the browser. For that, a new API endpoint has been created.

Fixes: #853

## Checklist before requesting review

- [x] Feature/fix PRs should add one atomic (as small as possible) feature or fix.
- [x] The code compiles and all the tests pass.

### CODE REVIEW

**Comment by ferran98campos** [comment]
How it used to look like: 
image

New version: 
image

File: api/views.py [hunk 1]
```diff
@@ -345,6 +346,48 @@ def organization(request):
...
+            companies = Company.objects.filter(name__icontains=input)[:limit]
+
+        data = [serializers.companies(request, company) for company in companies]
+        return JsonResponse(data, safe=False)
+
+    else:
+        return HttpResponseBadRequest("Unsupported method!", content_type="text/plain")
+
+
+def status(request):
```
Comment (didrikmunther) [inline api/views.py:371]: I remember deleting this function/endpoint, it must have gotten readded in this PR. Could you delete it?

File: docs/README.md [hunk 2]
```diff
@@ -70,14 +70,6 @@ After setting up the AIS with Docker, you can access it in a web browser with th
 
 4. Now you need to fill out some information. Fill out the necessary fields (Registration start date & end date, Complete registration start date & end date). Make sure end dates come after start dates. Tick the "Current" box, and press save at the bottom of the page. The fair will be created, and you can go back to `localhost:3000` to see the landing page for the fair.
 
-## Creating Database Migrations
```
Comment (didrikmunther) [inline docs/README.md:73]: Why is this part of the README being deleted?

File: make-migration.sh
Comment (didrikmunther) [inline make-migration.sh:1]: Why is this file being deleted?

File: templates/register/outside/create_user.html [hunk 3]
```diff
@@ -3,47 +3,99 @@
...
-				<h1>Create a new account</h1>
-			</div>
-		</div>
+    <div class="row header-row">
+      <div class="col-md-12 text-center">
+        <p class="h1">Create a new account</p>
+      </div>
+    </div>
 
+    <!--
```
Comment (didrikmunther) [inline templates/register/outside/create_user.html:26]: Why is this block commented out? Is there another place where you can create a new company? Can we remove the commented out code?

File: templates/register/outside/create_user.html [hunk 4]
```diff
@@ -3,47 +3,99 @@
...
+
+          <div class="form-step">{{ user_form | crispy }}</div>
+          <div class="col-md-12 text-center">
+            <button type="button" class="btn btn-default" id="back-step-btn">Back</button>
+            <button type="button" class="btn btn-default" id="next-step-btn">Next</button>
+            <button type="button" class="btn btn-success" id="submit-form-btn">Submit</button>
+          </div>
+
+          <!--
+						{% csrf_token %} {{ contact_form | crispy }}
```
Comment (didrikmunther) [inline templates/register/outside/create_user.html:91]: Why is this block commented out? Can we remove it?

**Review by didrikmunther — CHANGES_REQUESTED** [review]
I've left some nitpicks and general questions. A part of the README as well as an onboarding script is also missing, but overall I find this to be a very nice solution and I really love the commented code in the JS script. The new page looks fantastic!

File: api/views.py (hunk 1 above)
Comment (didrikmunther) [reply api/views.py:371]: No, I was referring to the `status` function on line 371.

**Review by didrikmunther — APPROVED** [review]
Very nice job!

**Pull request merged**